docs/runtime/go-metrics.md

```diff
@@ -0,0 +1,221 @@
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.limit(full) -->
+<!-- endsemconv -->
+
+### Metric: `go.memory.allocated`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.memory.allocated(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.allocated` | Counter | `By` | Memory allocated to the heap by the application. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/gc/heap/allocs:bytes`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.allocated(full) -->
+<!-- endsemconv -->
+
+### Metric: `go.memory.allocations`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.memory.allocations(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.allocations` | Counter | `{allocation}` | Count of allocations to the heap by the application. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/gc/heap/allocs:objects`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.allocations(full) -->
+<!-- endsemconv -->
+
+## Go Garbage Collection
+
+**Description:** Go metrics captured under the namespace `go.memory.gc.*`
+
+### Metric: `go.memory.gc.goal`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.memory.gc.goal(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.memory.gc.goal` | UpDownCounter | `By` | Heap size target for the end of the GC cycle. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/gc/heap/goal:bytes`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.memory.gc.goal(full) -->
+<!-- endsemconv -->
+
+## Go Goroutines
+
+**Description:** Go metrics captured under the namespace `go.goroutine.*`
+
+### Metric: `go.goroutine.count`
+
+This metric is [recommended][MetricRecommended].
+
+<!-- semconv metric.go.goroutine.count(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `go.goroutine.count` | UpDownCounter | `{goroutine}` | Count of live goroutines. [1] | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+
+**[1]:** Computed from `/sched/goroutines:goroutines`.
+<!-- endsemconv -->
+
+<!-- semconv metric.go.goroutine.count(full) -->
+<!-- endsemconv -->
+
+## Go Threads
+
+**Description:** Go metrics captured under the namespace `go.thread.*`
+
+### Metric: `go.thread.limit`
```

Comment:
I think this should be `go.procs.limit`:

1. This is not a limit on threads, but a limit on procs, which is an important difference. Calling this out in the metric description is too subtle IMO.
2. Diverging from the names chosen by `runtime/metrics` is a level of indirection that should be avoided.
3. What if Go decides to impose an actual thread limit in the future and export a metric for it?

Reply:
Ah, I had used this naming based on the description: "number of operating system threads" from https://pkg.go.dev/runtime/metrics.  Is `go.process.limit` OK?  It aligns better with otel conventions.

Reply:
`proc` is short for `processor` not `process`. So `go.processors.limit` would be okay I guess. But I think `procs` is the most common way to refer to this scheduler entity in the community.

Reply:
changed to processor for now. I notice the metric is named `/sched/gomaxprocs:threads`.  Does that mean the unit should be `{thread}`?

Reply:
> changed to processor for now. I notice the metric is named `/sched/gomaxprocs:threads`. Does that mean the unit should be `{thread}`?

I guess so. But I'd still stick with the term `processor` for the metric name.

Reply:
done